crates/transaction-generator-lib/src/workflow_delegator.rs

```diff
@@ -0,0 +1,239 @@
+                            0,
+                            1,
+                            Ordering::Relaxed,
+                            Ordering::Relaxed,
+                        );
+                        stage = 1;
+                    }
+                } else if stage < self.pool_per_stage.len()
+                    && self.pool_per_stage.get(stage - 1).unwrap().len() == 0
+                {
+                    info!("TransactionGenerator Workflow: Stage {} has consumed all accounts, moving to stage {}", stage, stage + 1);
+                    let _ = stage_counter.compare_exchange(
+                        stage,
+                        stage + 1,
+                        Ordering::Relaxed,
+                        Ordering::Relaxed,
+                    );
+                    stage += 1;
+                }
+            },
+            StageTracking::ExternallySet(_) => {
+                if stage == 0 {
+                    let prev = self
+                        .completed_for_first_stage
+                        .fetch_add(num_to_create, Ordering::Relaxed);
+                    num_to_create =
+                        cmp::min(num_to_create, self.num_for_first_stage.saturating_sub(prev));
+
+                    if num_to_create == 0 {
+                        return Vec::new();
+                    }
+                }
+            },
+        }
+
+        sample!(
+            SampleRate::Duration(Duration::from_secs(2)),
+            info!("Cur stage: {}, pool sizes: {:?}", stage, self.pool_per_stage.iter().map(|p| p.len()).collect::<Vec<_>>());
+        );
+
+        let result = if let Some(generator) = self.generators.get_mut(stage) {
+            generator.generate_transactions(account, num_to_create)
+        } else {
+            Vec::new()
+        };
+
+        result
+    }
+}
+
+pub struct WorkflowTxnGeneratorCreator {
+    stage: StageTracking,
+    creators: Vec<Box<dyn TransactionGeneratorCreator>>,
+    pool_per_stage: Vec<Arc<ObjectPool<LocalAccount>>>,
+    num_for_first_stage: usize,
+    completed_for_first_stage: Arc<AtomicUsize>,
+}
+
+impl WorkflowTxnGeneratorCreator {
+    fn new(
+        stage: StageTracking,
+        creators: Vec<Box<dyn TransactionGeneratorCreator>>,
+        pool_per_stage: Vec<Arc<ObjectPool<LocalAccount>>>,
+        num_for_first_stage: usize,
+    ) -> Self {
+        Self {
+            stage,
+            creators,
+            pool_per_stage,
+            num_for_first_stage,
+            completed_for_first_stage: Arc::new(AtomicUsize::new(0)),
+        }
+    }
+
+    pub async fn create_workload(
+        workflow_kind: WorkflowKind,
+        txn_factory: TransactionFactory,
+        init_txn_factory: TransactionFactory,
+        root_account: &mut LocalAccount,
+        txn_executor: &dyn ReliableTransactionSubmitter,
+        num_modules: usize,
+        _initial_account_pool: Option<Arc<ObjectPool<LocalAccount>>>,
```

Comment:
This input is not used. Are we planning to use it in the future?

Reply:
we do use it for the tournamnet